src/viam/sdk/robot/client.cpp

```diff
@@ -231,6 +236,62 @@ void RobotClient::refresh_every() {
     }
 };
 
+void RobotClient::check_connection() {
+    unsigned int check_every = check_every_interval_;
+    unsigned int reconnect_every = reconnect_every_interval_;
+    if (check_every == 0) {
+        check_every = reconnect_every;
+    }
+    if (check_every == 0 && reconnect_every == 0) {
+        should_check_connection_.store(false);
+    }
+    bool connected(true);
+    while (should_check_connection_) {
```

Comment:
@lia-viam another ignorant C++ question, `should_check_connection_` is set to false when we want to shutdown (makes sense!) but also above when `check_every` and `reconnect_every` are set to zero. In such a case this thread will return pretty much immediately, but the client code in such a case could be running for an indefinite amount of time. Are there any safety concerns here?